### PR TITLE
add user value to know if user signed up (no intent) via slack

### PR DESCRIPTION
**Pull request by camhashemi**
@leogrim @andrewconner @DerekBurch 

we need to show a popup after "normal" slack auth signup. this could probably be done several ways but decided not to get into the `intent` mess since this is a standard flow, now sets a `pref` in `finishSignup`. crossing my fingers.

alternatively, we could pass (then destroy a cookie) to the home page whenever someone completes signup, to do ftue-related stuff.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala [hunk 1]
```diff
@@ -252,6 +252,9 @@ class AuthHelper @Inject() (
               userCommander.sendWelcomeEmail(user.id.get, withVerification = !emailConfirmedAlready, Some(emailAddress))
             }
           }
+          authCommander.getUserIdentity(newIdentity.identityId).foreach {
+            case UserIdentity(richIdentity: SlackIdentity, _) => userValueRepo.setValue(user.id.get, UserValueName.SHOW_SLACK_CREATE_TEAM_POPUP, true)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala]: How does this work? Isn't this a match error if it's not a slack identity?

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala]: ahh, thank you. fixed it.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala [hunk 2]
```diff
@@ -252,6 +253,7 @@ class AuthHelper @Inject() (
               userCommander.sendWelcomeEmail(user.id.get, withVerification = !emailConfirmedAlready, Some(emailAddress))
             }
           }
+          if (slackMembershipRepo.getByUserId(user.id.get).nonEmpty) userValueRepo.setValue(user.id.get, UserValueName.SHOW_SLACK_CREATE_TEAM_POPUP, true)
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:256]: I might end up needing to go down the intent route after all, in dev mode I can't look at `UserIdentity` and know it is from Slack.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthHelper.scala:256]: this works in dev mode, deploying.

**Pull request merged**